tests/components/test_component_embedding.cpp

```diff
@@ -0,0 +1,52 @@
+#include <igl/is_edge_manifold.h>
+#include <catch2/catch_test_macros.hpp>
+#include <nlohmann/json.hpp>
+#include <wmtk_components/embedding/embedding.hpp>
+#include <wmtk_components/embedding/internal/Embedding.hpp>
+
+using json = nlohmann::json;
+using namespace wmtk;
+
+// this data is the one Prototype displayed
+void generateRandomData(Eigen::MatrixXd& vertices, Eigen::MatrixXi& edges, int vertices_num = 10)
+{
+    vertices.resize(vertices_num, 2);
+    edges.resize(vertices_num, 2);
+    std::srand((unsigned)time(NULL));
+    int intmax = std::numeric_limits<double>::lowest();
+    for (int i = 0; i < vertices_num; ++i) {
+        //rand()/double(RAND_MAX)
+        vertices(i, 0) = (std::rand() / double(intmax) - 0.5) * 10;
+        vertices(i, 1) = (std::rand() / double(intmax) - 0.5) * 10;
+        edges(i, 0) = i;
+        edges(i, 1) = (i + 1) % (vertices_num);
+    }
+}
+
+void generateDatawithSmallTriangles(Eigen::MatrixXd& vertices, Eigen::MatrixXi& edges)
+{
+    vertices.resize(10, 2);
+    edges.resize(10, 2);
+    vertices << 3.68892, 4.69909, 3.72799, -2.02109, 3.66695, -1.08875, -1.64113, 3.15912, -1.72872,
+        2.57591, 1.20228, 4.31211, 2.19108, -0.595569, 2.78527, -0.139927, -4.85534, 0.28489,
+        -4.68596, 3.57936;
+    edges << 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 0;
+}
+
+TEST_CASE("embedding_data_generation", "[components][embedding][2D]")
+{
+    Eigen::MatrixXd vertices;
+    Eigen::MatrixXi edges;
+    generateRandomData(vertices, edges);
+    generateDatawithSmallTriangles(vertices, edges);
```

Comment:
Would be nice if you add a function to write it to a file here.  You can comment it out because for the test it is not really necessary but it's good for having it here in case you want to check visually if everything is correct.

Reply:
I will fix it later(paste the edge_input code to this repository).